src/reports.rs

```diff
@@ -186,10 +203,16 @@ impl Reporter {
                 .and_then(|r| Some((r.original_response, r.attestation?)))
             {
                 const MAX_PAYLOAD_BYTES: usize = 100_000;
+                let allocation = if indexer_request.receipt.is_v1() {
+                    indexer_request.receipt.allocation().as_ref().to_vec() // Direct allocation for v1
+                } else {
+                    indexer_request.receipt.collection().as_address().to_vec() // Collection as address for v2
+                };
+
                 AttestationProtobuf {
                     request: Some(indexer_request.request).filter(|r| r.len() <= MAX_PAYLOAD_BYTES),
                     response: Some(original_response).filter(|r| r.len() <= MAX_PAYLOAD_BYTES),
-                    allocation: indexer_request.receipt.allocation().0.0.into(),
+                    allocation,
```

Comment:
So this @tmigone is being consumed by other components I think and so I'm being conservative about changing the interface and that's what I'm trying to get at in the comments.